nixos/modules/services/networking/nix-serve.nix

```diff
@@ -66,5 +66,7 @@ in
       description = "Nix-serve user";
       uid = config.ids.uids.nix-serve;
     };
+
+    nix.allowedUsers = [ "nix-serve" ];
```

Comment:
Note that setting this overrides the default value (`["*"]`) so it will actually lock out other users.